doc/tutorials/external-commands.md

```diff
@@ -0,0 +1,57 @@
+# How-to: Add external commands
+
+This tutorial will describe how to provide `kdb` with the specification of external programs.
+So `kdb` can parse and check the provided options and arguments according to the provided specification.
+This allows you to have, for example, a shell script but its args are checked by `kdb` before running it.
+It is possible to either mount(`kdb mount`) the specification, or set the keys manually using `kdb set` and `kdb meta-set`.
+Both options will be described in the following.
+
+For a reference of how the specification can look like [Command Line Options](command-line-options.md).
+
```

Comment:
Now you need to write what *specifically* you will do in this tutorial. It looks like this would be for something like `rm`? Please give a full example for how to write a specification for a useful script (e.g. doing `rm`).

I would prefer if it is an example for something related to Elektra.

Reply:
Added a small script that the tutorial is structured around. I couldn't really think of something that can't already be done by the normal commands.